### DISCUSSION TITLE
where is the openCursor from 1.6.3

### DISCUSSION THREAD

**Discussion by EliseoPlunker** [ujjwalguptaofficial/JsStore#228] [Q&A]
I has work with JsStore two year ago (ver. 1.6.3). Yes it is and old project that I want to upgrade to 4.2.6. Really I don't know if has any sense use cursors but this are dissapear in this new versions. There're a plan to implements or some work-around about it?
Thanks for your job and attention

**Comment by ujjwalguptaofficial** [comment]
Could you paste sample of your code @EliseoPlunker ? I am not able to recall if JsStore provided to use `cursor` .

**Reply by EliseoPlunker** [reply]
My god, sorry. You're correct. I manage the cursor using directly the indexedDB. Apologies and thanks for your time. I write the code below for if someone consider interesting (it's a simple extension of the example in mozilla page)

I needed show a loading with percent in a really large table. In desktop there're no problem, but some mobile spend a large time (8-10 seconds) and I don't wanted to use simply a rotating icon

       //in Angular I has a service dbsService
       public open(dbname:string): Observable<any> {
         return new Observable((observer: any) => {
           let request = indexedDB.open(dbname);

           request.onsuccess = () => {
             observer.next(request.result);
             observer.complete();
           }
           request.onerror = () => {
             alert(request.error);
             return throwError(request.error);
           }
         });
       }

       //and, use in a component
          this.data=[];
          this.dbsService.openCursor('indexedDB').subscribe((db: any) => {
            let tx = db.transaction("Table1", "readonly");
            let store = tx.objectStore("Table1");
            let index = store.index("index1");
            let request = index.openCursor(IDBKeyRange.lowerBound(lastItemID, true));
            request.onsuccess = () => {
              let cursor = request.result;
              if (cursor) {
                this.data.push({ ID: cursor.value.ID, Name: cursor.value.Name});
                let porc=Math.floor(cursor.value.ID*100/lastIndexID);
                console.log("starting... "+porc+"%");
                cursor.continue();
              } else {
                db.close();
                return of(true);
              }
            };
            db.onerror = (e: any) => {
              db.close();
              return throwError(e.target.error);
            }
          })
